deepspeed/runtime/engine.py

```diff
@@ -2444,7 +2455,9 @@ def load_checkpoint(self,
                         load_optimizer_states=True,
                         load_lr_scheduler_states=True,
                         load_module_only=False,
-                        custom_load_fn=None):
+                        custom_load_fn=None,
+                        enable_nebula_load=True,
```

Comment:
Will this not break backwards-compatibility by forcing existing load_checkpoint() calls in clients to use nebula? Can you clarify why nebula_* args need to be added to this function?

Reply:
Nope, it will not work. For case 2/3, it is doable in the side of 'initialize()'. But for case1, we must to have torch and nebula checkpoint engine ready to call `torch.load` and `nebula.save`.

It is hard for nebula to directly load the files saved by torch. Nebula maintained the customized checkpoint management services. 

Also, in real usage, if case 1/2/3 happen at one training process, it will be hard to directly call `deepspeed.initialize()` to make it ready.

Reply:
In that case, can the nebula checkpoint engine not be initialized for different load modes: (i) load with torch and (ii) load with nebula. This mode can be configured in the config. This will hide the complexity in the nebula checkpoint engine. How about this approach?

Reply:
Can you describe a situation, where cases 1, 2, and 3 will happen in a single training run?

Reply:
> Can you describe a situation, where cases 1, 2, and 3 will happen in a single training run?

I were thinking that the case where:
1. User want to load back the checkpoint saved by deepspeed when the job starts. (_case1_)
2. Let us say, after several steps(global_step1000), user's workflow will call `load_checkpoint` to load nebula checkpoint back for testing/validation. or something else, just to verify the status of saved checkpoints(_case2/3_).

In this case, the config would not work as it is one-off setting for training.
**But user must to change the code when they change the case from 1 to 2 or other cases.** :( . And the config can avoid this inconvenience. 

> In that case, can the nebula checkpoint engine not be initialized for different load modes: (i) load with torch and (ii) load with nebula. This mode can be configured in the config. This will hide the complexity in the nebula checkpoint engine. How about this approach?

Continue on the last one comment. Maybe we can assume user only `load_checkpoint` in a consistent case for a training. If so, as your suggestions, it would be better to put the arguments into configs. Any other concerns? If not, I can help move them into config. @tjruwase

Reply:
Thanks for explaining all the considerations. I think for now moving to config is sufficient. We can always address other user scenarios in the future when they come up. But for now, I think the simplicity of hiding the arguments in the config is priority.